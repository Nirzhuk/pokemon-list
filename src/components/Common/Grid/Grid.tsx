import styled from "styled-components";

export interface GridProps {
  width?: string;
  hasShadow?: boolean;
  margin?: string;
  centered?: boolean;
  grid?: string;
  gridRow?: string;
  gap?: string
  background?: boolean;
}

const Grid = styled.div<GridProps>`
  width: ${({ width }) => width};
  display: grid;
  grid-template-columns: ${({ grid }) => grid};
  grid-template-rows:${({ gridRow }) => gridRow};
  grid-gap: ${({ gap }) => gap};
  ${({ hasShadow }) =>
    hasShadow &&
    `
    border-radius: 8px;
    padding: 12px 8px;
    box-shadow: 0 0px 8px rgba(0, 0, 0, 0.2);`}
  ${({ margin }) => margin && `margin:${margin};`}
  ${({ centered }) => centered && `margin: 0 auto;`}
  ${({ background, theme }) => background && `background-color: ${theme.body}`}
`;

Grid.defaultProps = {
  width: "auto",
  hasShadow: false,
};

export default Grid;
