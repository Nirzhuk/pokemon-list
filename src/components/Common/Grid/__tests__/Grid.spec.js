import { render, screen } from '@testing-library/react'
import Grid from '../Grid'

describe('Grid', () => {
    const props = {
        grid: '1fr 1fr 1fr',
        gap: '25px'
    }
    const setup = (initialProps = props) => render(<Grid {...initialProps} />)
    it('Renders correctly', () => {
        const { container } = setup();

        expect(container).toMatchSnapshot();
    })
})