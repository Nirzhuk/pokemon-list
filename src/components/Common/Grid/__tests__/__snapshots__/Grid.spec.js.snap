// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Renders correctly 1`] = `
<div>
  <div
    class="sc-bcXHqe kgxZUT"
    width="auto"
  />
</div>
`;
